src/components/Table/DataTable/compenents/el-data-table/el-data-table.vue

```diff
@@ -1106,9 +1106,13 @@ export default {
       this.page = val
       this.getList()
     },
-    handleSelectAll(e, canSelect) {
-      this.loading = true
-      this.selectStrategy.onSelectAll(e, canSelect)
+    handleSelectAll(selection, selectable = () => true) {
+      this.tableLoading = true
+      try {
+        this.selectStrategy.onSelectAll(selection, selectable)
+      } finally {
+        this.tableLoading = false
+      }
     },
     /**
      * 切换某一行的选中状态，如果使用了第二个参数，则是设置这一行选中与否
```

Comment:
To summarize, I checked the given code without any irregularities identified based on current knowledge. There does not seem to be any known issues or significant optimization opportunities that could significantly enhance performance with current information.

In cases where certain parts might need tweaking due to their specific environment such as using dynamic data from a server vs loading it through local resources directly, these tweaks will depend heavily on specific requirements and constraints of the project.

The only suggestion would relate to updating some variable names which may lead to less confusing and descriptive usages:

```javascript
const end = (page + pageOffset) * size;
this.$log.debug(`page: ${page}, size: ${size}, start: ${start}, end: ${end}`);
```

This way, `page` becomes more generic than `index`, reducing its ambiguity towards users reading the code. But considering the context provided, no other suggested improvements were necessary for now.
  
Please let me know if you have another set of codes or questions about this one, please do share them!